src/acl/ConnMark.cc

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 1996-2018 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+/* DEBUG: section 28    Access Control */
+
+#include "squid.h"
+#include "acl/ConnMark.h"
+#include "acl/FilledChecklist.h"
+#include "cache_cf.h"
+#include "client_side.h"
+#include "Debug.h"
+#include "http/Stream.h"
+#include "HttpReply.h"
+#include "HttpRequest.h"
+#include "Parsing.h"
+#include "sbuf/Stream.h"
+
+#include <string>
+
+bool
+Acl::ConnMark::empty () const
+{
+    return false;
+}
+
+inline std::ostream &
+operator <<(std::ostream &os, const std::pair<nfmark_t,nfmark_t> connmark)
```

Comment:
Please consider defining that type because you use it quite a bit. For example:

```
/// a mark/mask pair for matching CONNMARKs
typedef std::pair<nfmark_t,nfmark_t> ConnMarkQuery;
```